inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/config/model/ClassInstrumentationConfiguration.java

```diff
@@ -16,23 +17,27 @@
     /**
      * The configuration representing that no instrumentation of the class is performed.
      */
-    public static final ClassInstrumentationConfiguration NO_INSTRUMENTATION = new ClassInstrumentationConfiguration(Collections.emptySet(), null);
+    public static final ClassInstrumentationConfiguration NO_INSTRUMENTATION = new ClassInstrumentationConfiguration(Collections.emptySet(), Collections.emptySet(), null);
 
     /**
      * The sensors which are active for the target class.
      */
     @Getter
     private final Set<SpecialSensor> activeSpecialSensors;
 
+    @Getter
+    private Set<InstrumentationRule> activeRules;
```

Comment:
Here I would recommend storing signatures of matched methods instead

Reply:
Related to the comment on equals and hashCode. Because these method should work as expected this should be fine.